### DISCUSSION TITLE
Invalid Linear system - SParseLDL Solver

### DISCUSSION THREAD

**Discussion by manueotti** [sofa-framework/sofa#3984] [Write / Run a simulation] [answered]
Hi,

I am trying to run a simulation of a hybrid soft finger that has a rigid tip. I have been facing a segmentation fault (SIGSEGV) error, accompanied by a warning related to the linear system solver. I would greatly appreciate your guidance in resolving this problem.

When I launch the simulation, there is no error message and the rigidification seems to have been made correctly. The segmentation fault happens at the start of the animation.

I use the SurfacePressureConstraint object in the direct control of pressure cavities. The following image shows the definition I used:
![image](https://github.com/sofa-framework/sofa/assets/102047172/5629dabe-9550-4ddc-bad3-fa5322beb97a)

Here is the image of the error message:
![image](https://github.com/sofa-framework/sofa/assets/102047172/bae60f80-f74a-441e-9742-aed0675afa1a)

**Comment by alxbilger** [comment]
Hi @manueotti 

You forgot to mention the version of SOFA you use.

Somehow, the linear solver fails to decompose the matrix of your problem. Check the geometry and topology to make sure that you don't end up with a 0x0 matrix or a an empty one.

I see that you also have other warnings. Warnings are important, and I advise you to fix them.

**Reply by manueotti** [reply]
Hi Alex,

Thanks for your reply. I'm using SOFA v22.06.

**Reply by alxbilger** [reply]
Ok. I believe that this segfault has been fixed in more recent versions of SOFA. However, you would still have the problem of the matrix decomposition.

**Reply by manueotti** [reply]
Hi, I have fixed the other warnings, but I can't seem to find a way to solve the problem with the matrix decomposition. Do you have an idea please?

**Reply by alxbilger** [reply]
You have to figure out why your linear system is either of size 0, or invalid (maybe empty). It can happen if you have 0 DoFs, or no contribution (no force, no mass etc). No contribution can happen if your topology is of size 0.

The picture you have included shows only the part of the scene that is mapped. Although mapped components contribute also to the linear system, they are not the main DoFs. You should also mention the main DoFs in this conversation.

**Reply by manueotti** [reply]
Thanks for your feedback, Alex. In my scene, I use a MechanicalMatrixMapper whose source object is the deformable part of the finger and the target object is the rigid part. 

I have a node for the finger model in which iI specify the mechanical properties and the degrees of freedom. This model node is parsed by the MechanicalMatrixMapper to perform the mapping from the deformable part to the rigid part. 

Following is a picture of the finger model containing the main DoFs:
![image](https://github.com/sofa-framework/sofa/assets/102047172/a8fe4dd4-6ce4-404d-b45f-b9c4497e01bc)

**Comment by alxbilger** [comment, marked as answer]
First weird thing: you have a second solver in the scene, at the level of the `model` Node. I don't think it is necessary.

I don't see any mapping between `DeformableMech` and `model/dofs`. Maybe your problem is due to this missing mapping.

Note that with the upcoming release v23.06, you will no longer need any MechanicalMatrixMapper.

**Reply by manueotti** [reply]
Hi Alex, thanks for your various responses. The problem has been solved.

1. I added a subsetMultiMapping to the model Node.
2.  I removed the linear solver in the model Node. (This removed the segmentation error).

Thank you very much for your attention,
Best regards.
